vars/MavenCommand.groovy

```diff
@@ -0,0 +1,142 @@
+def class MavenCommand {
+
+    def steps
+
+    boolean returnStdout = false
+
+    String settingsXmlPath = ''
+    List mavenOptions = []
+    Map properties = [:]
+    String logFileName = null
+    List profiles = []
+
+    MavenCommand(steps){
+        this.steps = steps
+    }
+
+    MavenCommand(steps, List defaultOpts){
+        this(steps)
+        this.mavenOptions.addAll(defaultOpts)
+    }
+
+    def run(String goals) {
+        def cmd = "mvn -B"
+        if(this.settingsXmlPath) {
```

Comment:
If, by mistake, we pass a null or empty string to the `withSettingsXmlFile` method, here we just skip the settings. Investigating such a bug can be very tedious, thus I would add fail-fast checks to every `with*` method.

Reply:
will do